### PR TITLE
Add skills button to attributes displays

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Add skills button to unit and group attributes displays to allow for fine-tuning of AI sub-skills
- Add `SLIDER:PERCENT` sub-type to the slider dialog control

### CODE REVIEW

File: addons/attributes/RscAttributes.hpp [hunk 1]
```diff
@@ -1057,6 +1061,12 @@ class GVAR(RscAttributesGroup): GVAR(RscAttributesBase) {
         class ButtonOK: ButtonOK {};
         class ButtonCancel: ButtonCancel {};
     };
+    class buttons {
```
Comment (CreepPork) [inline addons/attributes/RscAttributes.hpp]:
```suggestion
    class Buttons {
```

I'm not certain how arma reacts to this, but it looks better this way.

File: addons/attributes/functions/fnc_buttonSkills.sqf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * Example:
+ * [CONTROL] call zen_attributes_fnc_buttonSkills
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+private _entity = GETMVAR(BIS_fnc_initCuratorAttributes_target,objNull);
+private _unit   = if (_entity isEqualType grpNull) then {leader _entity} else {_entity};
```
Comment (CreepPork) [inline addons/attributes/functions/fnc_buttonSkills.sqf]: Maybe a select?

File: addons/attributes/RscAttributes.hpp (hunk 1 above)
Comment (mharis001) [reply addons/attributes/RscAttributes.hpp]: The config is not case-sensitive, I left it lowercase to match the other ones in display configs: `controls` and `controlsBackground`, but I agree it looks better capitalized 😄. Will change the others.

File: addons/attributes/functions/fnc_buttonSkills.sqf (hunk 2 above)
Comment (mharis001) [reply addons/attributes/functions/fnc_buttonSkills.sqf]: This is mostly personal preference, but `ARRAY select BOOL` does require building the array first. I think the rest of this component uses `if else` for attributes that work with multiple entity types.

**Pull request merged**